File: idea_app_backend/src/main/java/com/stitchagent/model/PulseMessage.java
package com.stitchagent.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PulseMessage {
    private String id;
    private String senderName;
    private String content;
    private String type; // USER, AGENT, EXPERT
    private String icon;
    private String codeSnippet;
    private String status; // For expert/agent progress
}

File: idea_app/lib/models/pulse_message.dart
class PulseMessage {
  final String id;
  final String senderName;
  final String content;
  final String type; // USER, AGENT, EXPERT
  final String icon;
  final String? codeSnippet;
  final String? status;

  PulseMessage({
    required this.id,
    required this.senderName,
    required this.content,
    required this.type,
    required this.icon,
    this.codeSnippet,
    this.status,
  });

  factory PulseMessage.fromJson(Map<String, dynamic> json) {
    return PulseMessage(
      id: json['id']?.toString() ?? '',
      senderName: json['senderName'] ?? '',
      content: json['content'] ?? '',
      type: json['type'] ?? 'USER',
      icon: json['icon'] ?? 'person',
      codeSnippet: json['codeSnippet'],
      status: json['status'],
    );
  }
}
